lib/client/api.go

```diff
@@ -2232,37 +2233,22 @@ func (tc *TeleportClient) connectToProxy(ctx context.Context) (*ProxyClient, err
 	}, nil
 }
 
-func makeProxySSHClientWithTLSWrapper(tc *TeleportClient, sshConfig *ssh.ClientConfig) (*ssh.Client, error) {
-	cfg := tc.Config
-	clientTLSConf, err := tc.loadTLSConfig()
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	clientTLSConf.NextProtos = []string{string(alpncommon.ProtocolProxySSH)}
-	clientTLSConf.InsecureSkipVerify = cfg.InsecureSkipVerify
-
-	tlsConn, err := tls.Dial("tcp", cfg.WebProxyAddr, clientTLSConf)
-	if err != nil {
-		return nil, trace.Wrap(err, "failed to dial tls %v", cfg.WebProxyAddr)
-	}
-	c, chans, reqs, err := ssh.NewClientConn(tlsConn, cfg.WebProxyAddr, sshConfig)
-	if err != nil {
-		// tlsConn is closed inside ssh.NewClientConn function
-		return nil, trace.Wrap(err, "failed to authenticate with proxy %v", cfg.WebProxyAddr)
-	}
-	return ssh.NewClient(c, chans, reqs), nil
-}
-
 func makeProxySSHClient(tc *TeleportClient, sshConfig *ssh.ClientConfig) (*ssh.Client, error) {
```

Comment:
I think if we refactor this a bit to keep the method that creates the ssh client over TLS tunnel like before, the code would be cleaner and easier to understand. Something like:

```go
func makeProxySSHClient(tc *TeleportClient, sshConfig *ssh.ClientConfig) (*ssh.Client, error) {
	if tc.Config.TLSRoutingEnabled {
		return makeProxySSHClientWithTLSWrapper(tc, sshConfig)
	}
	return makeProxySSHClientDirect(tc, sshConfig)
}

func makeProxySSHClientDirect(tc *TeleportClient, sshConfig *ssh.ClientConfig) (*ssh.Client, error) {
	dialer := proxy.DialerFromEnvironment(tc.Config.SSHProxyAddr, opts...)
	return dialer.Dial("tcp", tc.Config.SSHProxyAddr, sshConfig)
}

func makeProxySSHClientWithTLSWrapper(tc *TeleportClient, sshConfig *ssh.ClientConfig) (*ssh.Client, error) {
	clientTLSConf, err := tc.loadTLSConfig()
	if err != nil {
		return nil, trace.Wrap(err)
	}

	clientTLSConf.NextProtos = []string{string(alpncommon.ProtocolProxySSH)}
	clientTLSConf.InsecureSkipVerify = tc.Config.InsecureSkipVerify

	dialer := proxy.DialerFromEnvironment(tc.Config.WebProxyAddr,
		proxy.WithALPNDialer(), proxy.WithTLSConfig(tlsConfig))

	return dialer.Dial("tcp", tc.Config.WebProxyAddr, sshConfig)
}
```